### ISSUE TITLE
Adding a CUDA version to a toolchain

### ISSUE THREAD

**Issue by joeydumont** [ComputeCanada/software-stack#39]
Our cluster has CUDA 10.1 installed, and has been used by multiple users to compile their own code. I am working towards integrating the CC stack in this cluster. Would you recommend downgrading to CUDA 10.0 so that I can reuse packages built with the `gomklc-2018.3.312.100` toolchain, or is adding a new toolchain automated to a point that it would be worth it to add `gomklc-2018.3.312.101` and rebuild the packages needed?

Or is that needed? Is there a third option I'm missing?

Also, if this is not the proper channel for these kinds of questions, let me know.

**Comment by mboisson** [comment]
I would recommend installing the newer drivers for CUDA 11 and use the `2020a` version of the toolchains (including `gomklc-2020a`.

**Comment by joeydumont** [comment]
Ah, I was under the impression that StdEnv/2020 was not ready for production yet. I'll give that a whirl on my test nodes.

**Comment by mboisson** [comment]
It is not a full replacement yet, but most of the needs should be covered already. We have a lot of unused software packages under the other stacks.

**Comment by joeydumont** [comment]
This might be worth opening a separate issue, but I'm having issues with `whoami` on `StdEnv/2020`. whoami in the gentoo prefix does not seem to connect to the `nscd` daemon, and `strace` shows it trying to load `libnss_centrifydc.so.2`, but can't find it.

Do you know of a workaround on my end? Or is that an issue on the gentoo layer?

```
openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/etc/ld.so.cache", O_RDONLY\|O_CLOEXEC) = 3
--
  | mmap(NULL, 4096, PROT_READ\|PROT_WRITE, MAP_PRIVATE\|MAP_ANONYMOUS, -1, 0) = 0x7fffedb00000 |   | fstat(3, {st_mode=S_IFREG\|0644, st_size=187353, ...}) = 0
  | write(1, "admin.joey.dumont\n", 18)     = 18 |   | mmap(NULL, 187353, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fffedaac000
  |   |   | close(3)                                = 0
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/haswell/avx512_1/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/haswell/avx512_1/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/haswell/avx512_1/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/haswell/avx512_1", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/haswell/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/haswell/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/haswell/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/haswell", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/avx512_1/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/avx512_1/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/avx512_1/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/avx512_1", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/tls", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/haswell/avx512_1/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/haswell/avx512_1/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/haswell/avx512_1/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/haswell/avx512_1", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/haswell/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/haswell/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/haswell/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/haswell", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/avx512_1/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/avx512_1/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/avx512_1/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/avx512_1", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/lib64", {st_mode=S_IFDIR\|0755, st_size=4096, ...}) = 0
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/haswell/avx512_1/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/haswell/avx512_1/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/haswell/avx512_1/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/haswell/avx512_1", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/haswell/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/haswell/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/haswell/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/haswell", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/avx512_1/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/avx512_1/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/avx512_1/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/avx512_1", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/tls", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/haswell/avx512_1/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/haswell/avx512_1/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/haswell/avx512_1/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/haswell/avx512_1", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/haswell/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/haswell/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/haswell/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/haswell", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/avx512_1/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/avx512_1/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/avx512_1/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/avx512_1", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/x86_64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/x86_64", 0x7fffffff93c0) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64/libnss_centrifydc.so.2", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | stat("/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib64", {st_mode=S_IFDIR\|0755, st_size=4096, ...}) = 0
  |   |   | munmap(0x7fffedaac000, 187353)          = 0
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/etc/ld.so.cache", O_RDONLY\|O_CLOEXEC) = 3
  |   |   | fstat(3, {st_mode=S_IFREG\|0644, st_size=187353, ...}) = 0
  |   |   | mmap(NULL, 187353, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fffedaac000
  |   |   | close(3)                                = 0
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/libnss_files.so.2", O_RDONLY\|O_CLOEXEC) = 3
  |   |   | read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0003\0\0\0\0\0\0"..., 832) = 832
  |   |   | fstat(3, {st_mode=S_IFREG\|0755, st_size=51424, ...}) = 0
  |   |   | mmap(NULL, 54696, PROT_READ, MAP_PRIVATE\|MAP_DENYWRITE, 3, 0) = 0x7fffecf95000
  |   |   | mmap(0x7fffecf98000, 28672, PROT_READ\|PROT_EXEC, MAP_PRIVATE\|MAP_FIXED\|MAP_DENYWRITE, 3, 0x3000) = 0x7fffecf98000
  |   |   | mmap(0x7fffecf9f000, 8192, PROT_READ, MAP_PRIVATE\|MAP_FIXED\|MAP_DENYWRITE, 3, 0xa000) = 0x7fffecf9f000
  |   |   | mmap(0x7fffecfa1000, 8192, PROT_READ\|PROT_WRITE, MAP_PRIVATE\|MAP_FIXED\|MAP_DENYWRITE, 3, 0xb000) = 0x7fffecfa1000
  |   |   | close(3)                                = 0
  |   |   | mprotect(0x7fffecfa1000, 4096, PROT_READ) = 0
  |   |   | munmap(0x7fffedaac000, 187353)          = 0
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/etc/passwd", O_RDONLY\|O_CLOEXEC) = 3
  |   |   | lseek(3, 0, SEEK_CUR)                   = 0
  |   |   | fstat(3, {st_mode=S_IFREG\|0644, st_size=2249, ...}) = 0
  |   |   | read(3, "root:x:0:0:root:/root:/bin/bash\n"..., 4096) = 2249
  |   |   | lseek(3, 0, SEEK_CUR)                   = 2249
  |   |   | read(3, "", 4096)                       = 0
  |   |   | close(3)                                = 0
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/etc/ld.so.cache", O_RDONLY\|O_CLOEXEC) = 3
  |   |   | fstat(3, {st_mode=S_IFREG\|0644, st_size=187353, ...}) = 0
  |   |   | mmap(NULL, 187353, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fffedaac000
  |   |   | close(3)                                = 0
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/libnss_sss.so.2", O_RDONLY\|O_CLOEXEC) = 3
  |   |   | read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0@\25\0\0\0\0\0\0"..., 832) = 832
  |   |   | fstat(3, {st_mode=S_IFREG\|0755, st_size=37216, ...}) = 0
  |   |   | mmap(NULL, 2131056, PROT_READ\|PROT_EXEC, MAP_PRIVATE\|MAP_DENYWRITE, 3, 0) = 0x7fffecd8c000
  |   |   | mprotect(0x7fffecd94000, 2093056, PROT_NONE) = 0
  |   |   | mmap(0x7fffecf93000, 8192, PROT_READ\|PROT_WRITE, MAP_PRIVATE\|MAP_FIXED\|MAP_DENYWRITE, 3, 0x7000) = 0x7fffecf93000
  |   |   | close(3)                                = 0
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/lib64/libdl.so.2", O_RDONLY\|O_CLOEXEC) = 3
  |   |   | read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0000\21\0\0\0\0\0\0"..., 832) = 832
  |   |   | fstat(3, {st_mode=S_IFREG\|0755, st_size=14320, ...}) = 0
  |   |   | mmap(NULL, 16656, PROT_READ, MAP_PRIVATE\|MAP_DENYWRITE, 3, 0) = 0x7fffecd87000
  |   |   | mmap(0x7fffecd88000, 4096, PROT_READ\|PROT_EXEC, MAP_PRIVATE\|MAP_FIXED\|MAP_DENYWRITE, 3, 0x1000) = 0x7fffecd88000
  |   |   | mmap(0x7fffecd89000, 4096, PROT_READ, MAP_PRIVATE\|MAP_FIXED\|MAP_DENYWRITE, 3, 0x2000) = 0x7fffecd89000
  |   |   | mmap(0x7fffecd8a000, 8192, PROT_READ\|PROT_WRITE, MAP_PRIVATE\|MAP_FIXED\|MAP_DENYWRITE, 3, 0x2000) = 0x7fffecd8a000
  |   |   | close(3)                                = 0
  |   |   | mprotect(0x7fffecd8a000, 4096, PROT_READ) = 0
  |   |   | mprotect(0x7fffecf93000, 4096, PROT_READ) = 0
  |   |   | munmap(0x7fffedaac000, 187353)          = 0
  |   |   | openat(AT_FDCWD, "/var/lib/sss/mc/passwd", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/var/lib/sss/mc/passwd", O_RDONLY\|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  |   |   | getpid()                                = 562
  |   |   | fstat(-1, 0x7fffffff9bf0)               = -1 EBADF (Bad file descriptor)
  |   |   | getpid()                                = 562
  |   |   | socket(AF_UNIX, SOCK_STREAM, 0)         = 3
  |   |   | fcntl(3, F_GETFL)                       = 0x2 (flags O_RDWR)
  |   |   | fcntl(3, F_SETFL, O_RDWR\|O_NONBLOCK)    = 0
  |   |   | fcntl(3, F_GETFD)                       = 0
  |   |   | fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
  |   |   | connect(3, {sa_family=AF_UNIX, sun_path="/var/lib/sss/pipes/nss"}, 110) = -1 ENOENT (No such file or directory)
  |   |   | close(3)                                = 0
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/share/locale/locale.alias", O_RDONLY\|O_CLOEXEC) = 3
  |   |   | fstat(3, {st_mode=S_IFREG\|0644, st_size=2997, ...}) = 0
  |   |   | read(3, "# Locale name alias data base.\n#"..., 4096) = 2997
  |   |   | read(3, "", 4096)                       = 0
  |   |   | close(3)                                = 0
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/share/locale/en_US.UTF-8/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/share/locale/en_US.utf8/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/share/locale/en_US/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/share/locale/en.UTF-8/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/share/locale/en.utf8/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
  |   |   | openat(AT_FDCWD, "/cvmfs/soft.computecanada.ca/gentoo/2020/usr/share/locale/en/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
  |   |   | write(2, "whoami: ", 8)                 = 8
  |   |   | write(2, "cannot find name for user ID 171"..., 38) = 38
  |   |   | write(2, "\n", 1)                       = 1
```

**Comment by mboisson** [comment]
I created a new issue with this.

**Comment by joeydumont** [comment]
I will probably not be able to update to CUDA11 right away, as some machine learning packages do not support it yet.

Would it be possible for me to add support for CUDA10.1 while only having to recompile the CUDA-dependent packages?

**Comment by bartoldeman** [comment]
We have 10.2 (not 10.1) in the StdEnv/2020 stack but you need to load gcc/8.4.0 first (gcccuda-2019.6.102 toolchain, haven't made toolchains with MPI for that one yet).

For StdEnv/2018.3 we have the following toolchains:
for CUDA 10.2:
gcccuda 2019.3.102 (with GCC 8.3.0)
gcccuda 2018.3.102 (with GCC 7.3.0)

and for CUDA 10.1
gcccuda 2019.3.101 (GCC 8.3.0)
gcccuda 2018.3.101 (GCC 7.3.0)
is that sufficient or is something else still needed?

**Comment by mboisson** [comment]
Is the above sufficient @joeydumont  ?

**Comment by joeydumont** [comment]
Sorry about the delay, I am still not sure where the issue is. My original goal was to build [`ising-gpu`](https://github.com/NVIDIA/ising-gpu) with the easybuild in the CC stack. However, when trying to run the resulting binary, I get a CUDA driver mismatch:

```bash
$ cuIsing 

Using GPUs:
Cuda error in file 'main.cu' in line 1485 : CUDA driver version is insufficient for CUDA runtime version.
```

I can compile/run it fine outside of the CC stack. We have CUDA 10.1.105 with the 418.67 driver. I haven't found anything online that says that the CUDA toolkit 10.1.243 (the one in the CC stack) doesn't support that driver. I am not sure how to troubleshoot this further.

I also tried adding a CUDA/10.1.105-based gomklc toolchain, but I ran into issues with UCX (which I fixed by changing the LD_LIBRARY_PATH to point to the local `libibverbs.so`. Once that was fixed, I tried compiling OpenMPI-3.1.2-gcccuda-2018.3.101.105.eb (my new toolchain), but it seems that module hierarchy for CUDA takes only the major/minor portions of the version in consideration, so I tried to `rebuild` it, but I got a pretty weird error (at least, I never encountered it before):

```
**********************************************************************
* It appears that your C++ compiler is unable to link against object
* files created by your C compiler.  This generally indicates either
* a conflict between the options specified in CFLAGS and CXXFLAGS
* or a problem with the local compiler installation.  More
* information (including exactly what command was given to the
* compilers and what error resulted when the commands were executed) is
* available in the config.log file in this directory.
**********************************************************************
configure: error: C and C++ compilers are not link compatible.  Can not continue.
```

Right now I am trying to figure out what difference in the CC environment could lead to a driver mismatch error, as I think my original hypothesis that the CUDA toolkits were maybe incompatible might be flawed. If it's not flawed, adding the proper toolchain is taking a lot more effort than I thought it would. 

Anyway, sorry for the wall of text, hopefully you have some insight. Otherwise we can close this issue and I'll figure it out later, I pretty much answered the original question by trying it myself.

**Comment by joeydumont** [comment]
The issue with ising-gpu was caused by a busted CUDA install. A couple of symlinks later, I can run `cuIsing` built with the CC stack, with both my custom toolchain and the CC toolchain. I'll try to figure out the issue with OpenMPI, and if it doesn't work I'll open a separate issue.
